src/custom/hooks/useWalletInfo.ts

```diff
@@ -114,5 +114,5 @@ export function useWalletInfo(): ConnectedWalletInfo {
 export function useIsGnosisSafeApp(): boolean {
   const { walletName } = useWalletMetaData()
 
-  return walletName === GNOSIS_SAFE_APP_NAME
+  return !!walletName?.startsWith('Gnosis Safe')
```

Comment:
I know the reason for confusion.

This hook has a very specific usage: to identify if the app is loaded as a Gnosis Safe app.

But it has been used for a different purpose in the refactor https://github.com/cowprotocol/cowswap/blob/a585b3cd8ace69fff8368ef07e62da2183a5ffa4/src/custom/state/gnosisSafe/updater.tsx#L10

I would like to keep those usages separated.

We need a new hook just for this instead of re-using this one, probably `isGnosisSafeWallet` with your changes

Reply:
I've added an update, was that what you had in mind?

Reply:
Perfect, that's exactly it!